cli/internal/cmd/iamcreategcp.go

```diff
@@ -0,0 +1,188 @@
+	cmd.Flags().String("projectID", "", "ID of the GCP project the cluster will be deployed in. Find it on the welcome screen of your project: https://console.cloud.google.com/welcome")
+	must(cobra.MarkFlagRequired(cmd.Flags(), "projectID"))
+
+	return cmd
+}
+
+func runIAMCreateGcp(cmd *cobra.Command, args []string) error {
+	fileHandler := file.NewHandler(afero.NewOsFs())
+	spinner := newSpinner(cmd.ErrOrStderr())
+	defer spinner.Stop()
+	creator := cloudcmd.NewIAMCreator(spinner)
+
+	return iamCreateGcp(cmd, spinner, fileHandler, creator)
+}
+
+func iamCreateGcp(cmd *cobra.Command, spinner *spinner, fileHandler file.Handler, creator iamCreator) error {
+	// Get input variables
+	gcpFlags, err := parseGcpFlags(cmd)
+	if err != nil {
+		return err
+	}
+	cmd.Println(gcpFlags)
+
+	// verify CP auth
+	isAuthed, err := checkGcpAuth()
+	if err != nil {
+		return err
+	}
+	if !isAuthed {
+		cmd.PrintErrln("GCP authentication could not be verified. Please verify with the GCP CLI")
+	}
+
+	// Confirmation
+	cmd.Printf("The IAM configuration will be created on:\n")
+	cmd.Printf("Project ID:\t%s\n", gcpFlags.projectID)
+	cmd.Printf("Service Account ID:\t%s\n", gcpFlags.serviceAccountID)
+	cmd.Printf("Region:\t%s\n", gcpFlags.region)
+	cmd.Printf("Zone:\t%s\n", gcpFlags.zone)
+	ok, err := askToConfirm(cmd, "Do you want to create the configuration?")
+	if err != nil {
+		return err
+	}
+	if !ok {
+		cmd.Println("The creation of the configuration was aborted.")
+		return nil
+	}
+
+	// Creation
+	spinner.Start("Creating", false)
+	iamFile, err := creator.Create(cmd.Context(), cloudprovider.GCP, cloudcmd.IAMConfig{
+		GCP: cloudcmd.GCPIAMConfig{
+			ServiceAccountID: gcpFlags.serviceAccountID,
+			Region:           gcpFlags.region,
+			Zone:             gcpFlags.zone,
+			ProjectID:        gcpFlags.projectID,
+		},
+	})
+	spinner.Stop()
+	if err != nil {
+		return err
+	}
+
+	// Write back values
+	dec, err := base64.StdEncoding.DecodeString(iamFile.GCPOutput.ServiceAccountKey)
+	if err != nil {
+		return err
+	}
+
+	outJson := make(map[string]string)
+	err = json.Unmarshal([]byte(dec), &outJson)
+	if err != nil {
+		return err
+	}
+
+	cmd.Println("Service Account Key:", outJson["private_key_id"])
+	cmd.Println("Your IAM configuration was created successfully.")
+
+	return nil
+}
+
+// parseGcpFlags parses and validates the flags of the iam create gcp command
+func parseGcpFlags(cmd *cobra.Command) (gcpFlags, error) {
+	zone, err := cmd.Flags().GetString("zone")
+	if err != nil {
+		return gcpFlags{}, fmt.Errorf("parsing zone string: %w", err)
+	}
+	if !regexp.MustCompile(`^\w+-\w+-[abc]$`).MatchString(zone) { // Put into constants package later on
+		return gcpFlags{}, fmt.Errorf("invalid zone string: %s", zone)
+	}
+
+	region, err := cmd.Flags().GetString("region")
+	if err != nil {
+		return gcpFlags{}, fmt.Errorf("parsing region string: %w", err)
+	}
+	if region == "" {
+		// Infer region from zone
+		zoneParts := strings.Split(zone, "-")
```

Comment:
I probably dont get what strings.Cut does exactly here. From the docs I think it would cut a zone string, e.g. `"europe-west1-a"` (containing 2 dashes) at the first occurance of a dash, so I would be left with `["europe", "west1-a"]` (Which doesnt really make sense for the above usage?)

Reply:
You're right, missed that second dash. :sweat_smile: